arches/app/models/migrations/11613_nodegroup_root_node.py

```diff
@@ -0,0 +1,69 @@
+# Generated by Django 5.1.3 on 2024-11-19 09:29
+
+import django.db.models.deletion
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ("models", "10437_node_alias_not_null"),
+    ]
+
+    def add_root_node(apps, schema_editor):
+        NodeGroup = apps.get_model("models", "NodeGroup")
+        nodegroups_with_nodes = NodeGroup.objects.filter(node__gt=0).distinct()
+        for nodegroup in nodegroups_with_nodes:
+            nodegroup.root_node_id = nodegroup.pk
+        NodeGroup.objects.bulk_update(nodegroups_with_nodes, ["root_node_id"])
+
+        PublishedGraph = apps.get_model("models", "PublishedGraph")
+        published_graphs = PublishedGraph.objects.all()
+        for published_graph in published_graphs:
+            for node_dict in published_graph.serialized_graph["nodes"]:
+                node_dict["root_node_id"] = node_dict["nodegroup_id"]
+        PublishedGraph.objects.bulk_update(published_graphs, ["serialized_graph"])
+
+    def remove_root_node(apps, schema_editor):
+        PublishedGraph = apps.get_model("models", "PublishedGraph")
+        published_graphs = PublishedGraph.objects.all()
+        for published_graph in published_graphs:
+            for node_dict in published_graph.serialized_graph["nodegroups"]:
+                node_dict.pop("root_node_id", None)
+        PublishedGraph.objects.bulk_update(published_graphs, ["serialized_graph"])
+
+    operations = [
+        migrations.AddField(
+            model_name="nodegroup",
+            name="root_node",
```

Comment:
I'd argue just `node` here. `root_node` is less clear and can have different connotations.

Reply:
Happy to hold this until everyone's happy with the name, but `node` is the query name for a related item from `node_set`, so I don't think we can overload that. I think we need to land on something.